Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.45764&attempt_id=40c4f840-1d39-47cd-9587-c1902b976f7f)

Attempted: pango

<details><summary>Partial log (click to expand)</summary><p>

```
[69/129] Generating symbol file 'pango/pango@@pangocairo-1.0@sha/libpangocairo-1.0.0.dylib.symbols'.
[70/129] Generating PangoCairo-1.0.gir with a custom command.
FAILED: pango/PangoCairo-1.0.gir
/nix/store/ylclig98kn1j7q8b2bl3l5bnb4sc98yb-gobject-introspection-1.56.0-dev/bin/g-ir-scanner -I/nix/store/ylclig98kn1j7q8b2bl3l5bnb4sc98yb-gobject-introspection-1.56.0-dev/include/gobject-introspection-1.0 -I/nix/store/14ji0qzxppxbx24w2dccglpjvwchy60c-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/bysph12diipmacijqkm76fg869n740dg-glib-2.56.0/lib/glib-2.0/include --no-libtool --namespace=PangoCairo --nsversion=1.0 --warn-all --output pango/PangoCairo-1.0.gir --c-include=pango/pangocairo.h --quiet -DPANGO_ENABLE_BACKEND -DPANGO_ENABLE_ENGINE -I/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/pango -I/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango -I./. -I../. -I./pango/. -I../pango/. -I./pango/. -I../pango/. -I./pango/. -I../pango/. -I./pango/. -I../pango/. --filelist=/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango/pango@@pangocairo-1.0@sha/PangoCairo_1.0_gir_filelist --include-uninstalled=pango/Pango-1.0.gir --include=GObject-2.0 --include=cairo-1.0 --symbol-prefix=pango_cairo --identifier-prefix=PangoCairo --pkg-export=pangocairo --cflags-begin -D_POSIX_C_SOURCE=200809L -D_POSIX_THREAD_SAFE_FUNCTIONS -D_GNU_SOURCE -I./. -I../. -I./pango/. -I../pango/. -I/nix/store/14ji0qzxppxbx24w2dccglpjvwchy60c-glib-2.56.0-dev/include/glib-2.0 -I/nix/store/bysph12diipmacijqkm76fg869n740dg-glib-2.56.0/lib/glib-2.0/include -I/nix/store/9fcvvknzc56acknnbs4gpvxyj4zfi0s2-fribidi-1.0.5/include/fribidi -I/nix/store/g8bgy3i1423ffz65hlrnk0fzv56s34qc-harfbuzz-1.8.8-dev/include/harfbuzz -I/nix/store/gjp30amwqi242csh0rmj3iz482f59lk4-fontconfig-2.12.6-dev/include -I/nix/store/pqgilyfdkr9n5fl613s1brfjmw0511wr-expat-2.2.6-dev/include -I/nix/store/ys684rkpv2kiwldp63bxk4jyrwyk76g2-freetype-2.9-dev/include/freetype2 -I/nix/store/89xpcjfr4gd0x2dm1y7ajx71hmb0lh8n-xproto-7.0.31/include -I/nix/store/w7mhbz22z5jm1m3xq2vwm0pl8j62yvac-libXrender-0.9.10-dev/include -I/nix/store/5qm0mmhzb6j33qdr6644vq7fz3vbirpy-renderproto-0.11.1/include -I/nix/store/wpv5m1b54fhv0ad3xbvcqvf0j34ns40s-libX11-1.6.6-dev/include -I/nix/store/vlxmahi3mbq4hvddnxb9ql90dy8krl3h-kbproto-1.0.7/include -I/nix/store/nix00lgaprlpaxpcg29yrxjfx99whivz-libXft-2.3.2-dev/include -I/nix/store/fcs8183pyn7akshb1ykn91wxmqh412x0-cairo-1.15.12-dev/include/cairo -I/nix/store/ys684rkpv2kiwldp63bxk4jyrwyk76g2-freetype-2.9-dev/include --cflags-end -L/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango --extra-library=m -L/nix/store/bysph12diipmacijqkm76fg869n740dg-glib-2.56.0/lib --extra-library=gobject-2.0 --extra-library=glib-2.0 -L/nix/store/9fcvvknzc56acknnbs4gpvxyj4zfi0s2-fribidi-1.0.5/lib --extra-library=fribidi -L/nix/store/kwzlrnldqzch3z46rajix3wjdxfzhiv1-harfbuzz-1.8.8/lib --extra-library=harfbuzz -L/nix/store/jl2j3xd11v3alimvfx0zg7qfk8gdx3sh-fontconfig-2.12.6-lib/lib -L/nix/store/k1w4sfd66mf7nslibwrkfxw7ycpwgll0-freetype-2.9/lib --extra-library=fontconfig --extra-library=freetype -L/nix/store/biihf7k6fwr5gy59f8n47jrmn7zmz4z1-libXrender-0.9.10/lib -L/nix/store/r4f1w1sqm04yxhm7bk74fas9ifz0k79v-libX11-1.6.6/lib --extra-library=Xrender --extra-library=X11 -L/nix/store/0mgpdp3hqrlixiqm0z2n7nwl10qsyw18-libXft-2.3.2/lib --extra-library=Xft -L/nix/store/ys1sp7dmcibcii63hmdgi5wdfyc1m5l6-cairo-1.15.12/lib --extra-library=cairo -framework CoreFoundation ApplicationServices -L/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango --extra-library=pango-1.0 --extra-library=pangoft2-1.0 --add-include-path=/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango --add-include-path=/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango -L/private/tmp/nix-build-pango-1.42.4.drv-0/pango-1.42.4/build/pango --library pangocairo-1.0
Usage: g-ir-scanner [options] sources

g-ir-scanner: error: no such option: -f
ninja: build stopped: subcommand failed.
builder for '/nix/store/i3pbz62fwck43lxlryrlrkxlq94d2d03-pango-1.42.4.drv' failed with exit code 1
error: build of '/nix/store/i3pbz62fwck43lxlryrlrkxlq94d2d03-pango-1.42.4.drv' failed
```
</p></details>

--- a/pkgs/development/libraries/pango/default.nix
+++ b/pkgs/development/libraries/pango/default.nix
@@ -1,37 +1,45 @@
 { stdenv, fetchurl, pkgconfig, libXft, cairo, harfbuzz
-, libintl, gobjectIntrospection, darwin, fribidi
+, libintl, gobjectIntrospection, darwin, fribidi, gnome3
+, gtk-doc, docbook_xsl, docbook_xml_dtd_43, makeFontsConf, freefont_ttf
 }:
 
 with stdenv.lib;
 
 let
-  ver_maj = "1.42";
-  ver_min = "1";
-in
-stdenv.mkDerivation rec {
-  name = "pango-${ver_maj}.${ver_min}";
+  pname = "pango";
+  version = "1.42.4";
+in stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
 
   src = fetchurl {
-    url = "mirror://gnome/sources/pango/${ver_maj}/${name}.tar.xz";
-    sha256 = "0cnfgcya3wbs9m8g44cl5ww6wbp6qbw96qvsgkr8ymwqn9b6fnli";
+    url = "mirror://gnome/sources/${pname}/${stdenv.lib.versions.majorMinor version}/${name}.tar.xz";
+    sha256 = "17bwb7dgbncrfsmchlib03k9n3xaalirb39g3yb43gg8cg6p8aqx";
   };
 
   outputs = [ "bin" "dev" "out" "devdoc" ];
 
-  buildInputs = [ gobjectIntrospection ];
-  nativeBuildInputs = [ pkgconfig ]
-    ++ optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
-       Carbon
-       CoreGraphics
-       CoreText
-    ]);
+  nativeBuildInputs = [ pkgconfig gobjectIntrospection gtk-doc docbook_xsl docbook_xml_dtd_43 ];
+  buildInputs = optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+    Carbon
+    CoreGraphics
+    CoreText
+  ]);
   propagatedBuildInputs = [ cairo harfbuzz libXft libintl fribidi ];
 
   enableParallelBuilding = true;
 
-  configureFlags = optional stdenv.isDarwin "--without-x";
+  # Fontconfig error: Cannot load default config file
+  FONTCONFIG_FILE = makeFontsConf {
+    fontDirectories = [ freefont_ttf ];
+  };
 
-  doCheck = false; # fails 1 out of 12 tests with "Fontconfig error: Cannot load default config file"
+  doCheck = false; # /layout/valid-1.markup: FAIL
+
+  passthru = {
+    updateScript = gnome3.updateScript {
+      packageName = pname;
+    };
+  };
 
   meta = with stdenv.lib; {
     description = "A library for laying out and rendering of text, with an emphasis on internationalization";

